src/test/java/net/kaczmarzyk/spring/data/jpa/web/SpecificationArgumentResolverTest.java

```diff
@@ -214,6 +233,14 @@ public static interface CustomSpecRepeatedFetch extends Specification<Object> {
     @Spec(path = "path1", spec = Like.class)
     public static interface CustomSpecJoinContainer extends Specification<Object> {
     }
+
+    @net.kaczmarzyk.spring.data.jpa.web.annotation.Conjunction(
+            @Or(
+                    @Spec(path = "path1", spec = Like.class)
+            )
+    )
+    public static interface ConjunctionSpec extends Specification<Object> {
+    }
     
     public static class TestController {
```

Comment:
add additional tests specific to proxy strategy that you changed (in a separate test file):
* method parameter type is Specification and there is only one inner spec (via @Spec) = no proxy is created 
* method parameter type is Specification and there are multiple inner specs (@Join + @Spec) = no proxy is created 
* method parameter type is Conjunction (multiple inner specs) = no proxy is created 
* method parameter type is Like and there is only one inner spec (Like created with @Spec = no proxy is created
* method parameter type is a custom interface = proxy is created

+ add assertion that proper type was returned in each case